src/main/java/org/jenkinsci/plugins/ewm/steps/ExwsAllocateExecution.java

```diff
@@ -91,4 +126,42 @@ private String computePathOnDisk(String physicalPathOnDisk) {
         FilePath diskFilePath = new FilePath(new File(physicalPathOnDisk));
         return new FilePath(diskFilePath, run.getParent().getFullName() + "/" + run.getNumber()).getRemote();
     }
+
+    /**
+     * Find the upstream job that matches the given upstream name.
+     *
+     * @param upstreamJob the upstream job name
+     * @return the upstream job with the given name if any, {@code null} otherwise
+     */
+    @CheckForNull
+    private Item findUpstreamJob(String upstreamJob) {
```

Comment:
I think the more general approach is better: the upstream/downstream jobs may not be in the same folder. Oleg said something about relative job names and absolute job names, and complications with computing jobs names (maybe he was thinking about the "job" string being inserted between folder names and job names - but he did not say this explicitly so he may have meant something else).